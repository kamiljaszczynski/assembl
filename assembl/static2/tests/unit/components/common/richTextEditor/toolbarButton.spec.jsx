import React from 'react';
import renderer from 'react-test-renderer';

import ToolbarButton from '../../../../../js/app/components/common/richTextEditor/toolbarButton';

describe('ToolbarButton component', () => {
  const onToggleSpy = jest.fn(() => {});
  it('should render an active toolbar button', () => {
    const props = {
      isActive: true,
      icon: 'text-bold',
      label: 'Bold',
      onToggle: onToggleSpy
    };
    const component = renderer.create(<ToolbarButton {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an inactive toolbar button', () => {
    const props = {
      isActive: false,
      icon: 'text-italic',
      label: 'Italic',
      onToggle: onToggleSpy
    };
    const component = renderer.create(<ToolbarButton {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});