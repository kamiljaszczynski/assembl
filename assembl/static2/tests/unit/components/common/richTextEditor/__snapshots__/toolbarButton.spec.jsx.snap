// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton component should render an active toolbar button 1`] = `
<button
  className="active btn btn-default"
  disabled={false}
  onClick={[Function]}
  title="Bold"
  type="button"
>
  <span
    className="assembl-icon-text-bold"
  />
</button>
`;

exports[`ToolbarButton component should render an inactive toolbar button 1`] = `
<button
  className="btn btn-default"
  disabled={false}
  onClick={[Function]}
  title="Italic"
  type="button"
>
  <span
    className="assembl-icon-text-italic"
  />
</button>
`;
